### PR TITLE
s4583670 Vision Transformer for Alzheimer's Classification

### PR DESCRIPTION
**Pull request by ryanlederhose**
Hi Teaching Team,

Please see my completed vision transformer model for classifying Alzheimer's disease on the ADNI brain dataset. The model performed with a test accuracy of `67.65%`, which is slightly lower than I would have liked. I split the training dataset into a train/validation split using an 80/20 ratio. The split also uses a patient-level split to avoid data leakage. One particularly noteworthy part of my design was to use a convolutional layer for both feature extraction and to split the input image into patches. I didn't find any pre-processing methods particularly helped the accuracy of my design. For an improved output, I really needed a bigger dataset.

Thankyou,
Ryan - 45836705

### CODE REVIEW

**Comment by LinfengLiu98** [comment]
**This is an initial inspection, no action is required at this point**

Difficulty: Hard

Readme:
Excellent 

Commit messages: Excellent, detailed.

Code:
- Uses Vision transformer
- Well commented
- Good model design

Functionality/Performance:
- Test accuracy is 67.65%
- Preprocssing data correctly
- No data augmentation
- Well done

General comments:
Can perform data augmentation to prevent overfitting.

**Comment by shakes76** [comment]
# Marking
## Good Practice (Design/Commenting, TF/Torch Usage)
Adequate design and implementation
Good spacing and comments
Header blocks
## Recognition Problem 
Solves problem, no augmentation -1
Driver Script present
File structure present
Shows Usage & Demo & Visualisation & Data usage
Module present
Commenting
No Data leakage
Difficulty: Hard
## Commit Log
Meaningful commit messages sometimes -1
Progressive commits used
## Documentation
ReadMe good, no refs -1
Model/technical explanation
Good Description and Comments 
Markdown used and PDF submitted
## Pull Request
Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch) 
No Feedback required
Request Description good
